lib/kubernetes-deploy/resource_cache.rb

```diff
@@ -51,9 +51,10 @@ def use_or_populate_cache(kind)
 
     def fetch_by_kind(kind)
       resource_class = KubernetesResource.class_for_kind(kind)
+      global_kind = @task_config.global_kinds.include?(kind.downcase)
```

Comment:
Seeing this, I definitely think that `TaskConfig` should do the fetching. Otherwise we have a call ordering dependency involving three objects (this, the task and the mediating task config).